### PR TITLE
MB-11579 framework for formatting history data details

### PR DESCRIPTION
**Pull request by YanZ777**
## Summary

Created framework for how the move details will formatted.
This mainly deals with formatting the `Updated Move` and `Approved Move`, which should display `Move flagged for financial review` or `Move unflagged for financial review` and `Created Move Task Order (MTO)`.

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Start the UI locally.

```sh
make office_client_run
```

##### Terminal 3

Start the Go server locally.

```sh
make server_run
```

</details>

### Additional steps

1. Access the office application
2. Login as a TOO
3. Select a move with status "New Move"
4. Approve some shipments on the Move Details Page to approve a move task order 
5. On the Move Task Order Page, flag move for financial review and unflag it
6. Go to the Move History Page and see that the details column shows the correct information

## Verification Steps for Author

These are to be checked by the author.

- [ ] Request review from a member of a different team.
- [x] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output

## Screenshots

image
image

### CODE REVIEW

**Review by juliemoon — APPROVED** [review]
This looks great! I included a screenshot of what I'm seeing
Screen Shot 2022-03-25 at 10 06 57 AM
.

**Review by michaelvaldes-truss — APPROVED** [review]
LGTM! I love that you added a middle step for the MoveHistoryDetailsSelector

File: src/constants/historyLogUIDisplayName.js [hunk 1]
```diff
@@ -1,4 +1,103 @@
-const historyLogEventNameDisplay = {
+import PropTypes from 'prop-types';
+
+export const HistoryLogValuesShape = PropTypes.arrayOf(
+  PropTypes.shape({
+    columnName: PropTypes.string.isRequired,
+    columnValue: PropTypes.string,
+  }),
+);
```
Comment (rogeruiz) [inline src/constants/historyLogUIDisplayName.js:8]: We will need to revisit this as we move forward with the discussions happening in Confluence.

[=> Data serialization and deserialization for history/audit logging 🔒 ](https://dp3.atlassian.net/wiki/spaces/MT/pages/1698922499/Data+serialization+and+deserialization+for+history+audit+logging)

File: src/pages/Office/MoveHistory/MoveHistoryDetailsSelector.jsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+const MoveHistoryDetailsSelector = ({ eventName, oldValues, changedValues }) => {
+  /**
+   * Inside the component, we should map oldValues and changedValues into an object so the ordering can be consistent.
+   */
+  if (eventNamesWithLabelledDetails[eventName]) {
+    return (
+      <div>
+        Labelled {eventName}
```
Comment (donaldthai) [inline src/pages/Office/MoveHistory/MoveHistoryDetailsSelector.jsx:30]: nit: Same in `historyLogUIDisplayName.js`, `Labeled` is the one to use in American context.

File: src/pages/Office/MoveHistory/MoveHistoryDetailsSelector.test.jsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      'Move flagged for financial review',
+      [{ columnName: 'financial_review_flag', columnValue: 'true' }],
+    ],
+    [
+      'setFinancialReviewFlag',
+      'Move unflagged for financial review',
+      [{ columnName: 'financial_review_flag', columnValue: 'false' }],
+    ],
+    ['updateMoveTaskOrderStatus', 'Created Move Task Order (MTO)', [{ columnName: 'status', columnValue: 'APPROVED' }]],
+    ['updateMoveTaskOrderStatus', 'Created Move Task Order (MTO)', [{ columnName: 'status', columnValue: 'APPROVED' }]],
```
Comment (donaldthai) [inline src/pages/Office/MoveHistory/MoveHistoryDetailsSelector.test.jsx:46]: This seems to be a duplicate?

**Review by donaldthai — APPROVED** [review]
Left some comments that are nits but this looks great! 🚀

**Pull request merged**